Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -80,9 +85,11 @@ public ConnectionPoolStatus Status
             Uri uri,
             IPooledConnectionFactory connectionFactory,
             ConnectionPoolSettings connectionPoolSettings,
-            ILogger logger)
-            : base(logger)
+            ILogging logging)
         {
+            _id = $"Pool-{UniqueIdGenerator.GetId()}";
```

Comment:
What about using `hostname:port` pair as an identifier here? Maybe we can work on existing log calls as another PR later on and provide helpful context?